app/coffee/Features/Labels/LabelsHandler.coffee

```diff
@@ -0,0 +1,50 @@
+ProjectEntityHandler = require "../Project/ProjectEntityHandler"
+DocumentUpdaterHandler = require('../DocumentUpdater/DocumentUpdaterHandler')
+
+
+module.exports = LabelsHandler =
+
+	labelCaptureRegex: () ->
+		/\\label\{([^\}\n\\]{0,80})\}/g
+
+	getAllLabelsForProject: (projectId, callback=(err, projectLabels)->) ->
+		ProjectEntityHandler.getAllDocs projectId, (err, docs) ->
+			if err?
+				return callback(err)
+			LabelsHandler.extractLabelsFromProjectDocs docs, (err, projectLabels) ->
+				if err?
+					return callback(err)
+				callback(null, projectLabels)
+
+	getLabelsForDoc: (projectId, docId, callback=(err, docLabels)->) ->
+		DocumentUpdaterHandler.flushDocToMongo projectId, docId, (err) ->
```

Comment:
`getLabelsForDoc` flushes first, but `getAllLabelsForProject` doesn't. Is this correct, if so can you add a comment to explain why.